### ISSUE TITLE
to Enoki page compatibility

### ISSUE THREAD

**Issue by kodedninja** [kodedninja/solo#1]
A place to throw ideas. @jondashkyle

**Comment by jondashkyle** [comment]
awesome. so as you mentioned, enoki constructs it’s content object (with the`hypha` module) by traversing directories containing `index.txt` files. the directory name and it’s relation to the root directory becomes the pathname. this structure is conducive to placing media related to a page alongside it’s content by including it in the same directory the `index.txt` file exists; ultimately a structure that scales quite well.

solo is simpler in it’s structure. you can place multiple `.txt` files within a directory, each becoming a path. this results in a much flatter directory structure. orkl adds another layer on top of this, which we should support too!

one way of supporting both taxonomies is by building this ability into hypha. it could determine the structure by reading the structure, and not explicitly through an option. for instance:

1. if there is an` index.txt`, read enoki style
2. if there is a single `.txt` file which is not meta information for a file (e.g. `image.png.txt`), treat that as the `index.txt`
3. if there are multiple `.txt` files which are not meta information for a file, treat as sub-directories and read the content as an `index.txt` file.

where this gets a little tricky is writing that data back to the archive. for example, if you opened a solo site in enoki, you should be able to save new content in addition to reading the content into the interface. perhaps there is a utility function which determines the taxonomy when writing data?

basically; i’m wanting to avoid options. it should just work by default. what could be nice is to build better tests into hypha. you can define a new taxonomy in hypha and write a test for it. so rather than trying to create a single data structure, we simply ensure you can get from one to another easily.

**Comment by kodedninja** [comment]
Tell me if I don't understand you, but I wrote about orkl (the blogging site) and not solo (the 1 text per link thing).

Did I misunderstand you? :smile:
